internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -440,6 +441,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		cds[i] = composedRendered{
 			resource: cd,
 			rendered: rendered,
+			filteredPatches: filterPatches(ta.Template.Patches, v1.PatchTypeFromCompositeFieldPath,
```

Comment:
We will need to maintain this as new patch types are added.

Reply:
Perhaps we should use "constant" package level variables with docstrings to help make it more obvious that we need to keep these up to date? e.g.:

```go
var (
	// Types of patches that are _from_ a composite resource to a composed resource.
	fromXRTypes = []v1.PatchType{...}

	// Types of patches that are from a composed resource _to_ a composite resource.
	toXRTypes = []v1.PatchType{...}
)
```

Reply:
Done. Added `composed.patchTypesToXR` & `composed.patchTypesFromXR` functions to define XR-targeting & composed-targeting patch types, respectively.